docs/sources/tempo/configuration/_index.md

```diff
@@ -489,6 +489,10 @@ querier:
         # The default value of "" disables this feature.
         [external_endpoints: <list of strings> | default = <empty list>]
 
+        # If external_backend is set, then authorization credentials will be provided when querying the external endpoints. "cloud-run" is the only value supported at this time.
+        # The default value of "" omits credentials when querying the external backend.
+        [external_backend: <string> | default = ""]
```

Comment:
reading the code below i feel like this should be "external_backend_auth"? 

or a slightly different direction:

i'm wondering if longterm we'd prefer a pattern like we do in the storage backend:

```
external_backend: <google-cloud-run|aws-lambda|http>
google-cloud-run:
  <gcr settings>
aws-lambda:
  <lambda settings>
...

Reply:
Sounds good! I pushed up a change that restructures the config with an eye toward these longer-term considerations. I'm hoping the new structure will avoid breaking changes in the future, while providing enough flexibility to define the cloud-run/lambda/http specific as we discover them. For now, I'm holding off on making any breaking changes in the existing config as well.

Reply:
Just to clarify, the SearchConfig looks like this now:

```
// These are the existing configs:
external_endpoints: []string
HedgeRequestsAt: int
HedgeRequestsUpTo: int
// ^ moving these under their own "http" external_backend block is a followup,
// requiring a breaking change

external_backend: <google_cloud_run> // "aws-lambda" and "http" will be a followup
google_cloud_run:
  external_endpoints: []string // eg: ["https://endpoint1.run.app", "https://endpoint2.run.app" ]
  HedgeRequestsAt: int
  HedgeRequestsUpTo: int
```

There's some repetition in the `external_endpoints` and `HedgeRequests*` configs for each external backend type, but I figured it made more sense to make that declarative instead of trying to re-use it across multiple backends. This declarative approach is also consistent with the storage backend config [`/tempodb/config.go`](https://github.com/grafana/tempo/blob/d6544e80c076a3b96a0c991bc329252c926ffede/tempodb/config.go#L52-L57).

Also, I kept `external_endpoints` as a list under the `google_cloud_run` config. I'm not totally sure why it's a list to begin with - perhaps we want to load balance across multiple serverless deployments or regions? I figured that we'll want to retain the list, but I'm calling that out here just to be sure.

Reply:
I like the direction this is heading. See some more detailed comments below.